File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    math1 = [1, 2, 3, 4, 5]
    math1_scr = 0
    math2 = [2, 1, 2, 3, 2, 4, 2, 5]
    math2_scr = 0
    math3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    math3_scr = 0
    result = []
    for idx, answer in enumerate(answers):
        if math1[idx%5] == answer:
            math1_scr += 1
        if math2[idx%8] == answer:
            math2_scr += 1
        if math3[idx%10] == answer:
            math3_scr += 1
    max_scr = max(math1_scr, math2_scr, math3_scr)
    if math1_scr == max_scr:
        result.append(1)
    if math2_scr == max_scr:
        result.append(2)
    if math3_scr == max_scr:
        result.append(3)
    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[3];
        List<Integer> arraylist = new ArrayList<>();
        for (int i = 0; i < answers.length; i++) {
            if (a[i%a.length] == answers[i]) {
                score[0]++;
            } if (b[i%b.length] == answers[i]) {
                score[1]++;
            } if (c[i%c.length] == answers[i]) {
                score[2]++;
            }
        }
        int maxScore = Math.max(score[0], Math.max(score[1], score[2]));
        if (score[0] == maxScore) {
            arraylist.add(1);
        } if (score[1] == maxScore) {
            arraylist.add(2);
        } if (score[2] == maxScore) {
            arraylist.add(3);
        }
        int[] answer = new int[arraylist.size()];
        for (int i = 0; i < arraylist.size(); i++) {
            answer[i] = arraylist.get(i);
        }
        return answer;
    }
}